src/sagas/deposit.js

```diff
@@ -301,6 +304,50 @@ export function* getBitcoinAddress() {
   yield* autoSubmitDepositProof()
 }
 
+function createConfirmationChannel(deposit) {
+  return eventChannel((emit) => {
+    const listener = ({
+      transactionID,
+      confirmations,
+      requiredConfirmations,
+    }) => {
+      emit({ transactionID, confirmations, requiredConfirmations })
+
+      // Close channel once we have all the required confirmations
+      if (confirmations === requiredConfirmations) {
+        emit(END)
+      }
+    }
+
+    deposit.onReceivedFundingConfirmation(listener)
+
+    // no-op
+    // eventChannel subscriber must return an unsubscribe, but tbtc.js does not
+    // currently provide one
+    return () => {}
+  })
+}
+
+function* watchForFundingConfirmations(deposit) {
+  yield delay(500)
+
+  const confirmationChannel = yield call(createConfirmationChannel, deposit)
+  try {
+    while (true) {
+      const { transactionID, confirmations } = yield take(confirmationChannel)
+      yield delay(50)
+      if (confirmations) {
+        yield put({
+          type: BTC_TX_CONFIRMED,
+          payload: { btcConfirmedTxID: transactionID, confirmations },
+        })
+      }
+    }
+  } finally {
+    console.debug("And now, the watch has ended")
```

Comment:
Is this because `take` will `throw` when the channel closes?

Reply:
Tbh I was interested in watching for debug statement. 😅 We can just remove it - without it the saga should just end.